service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaService.java

```diff
@@ -79,27 +80,41 @@ public Policy enablePetServiceAccountImpersonation(
    *
    * @param workspaceUuid ID of the workspace to enable pet SA in
    * @param userToEnableEmail The user whose proxy group will be granted permission.
-   * @param token Token for calling SAM.
+   * @param userReq Auth token for calling SAM.
    * @param eTag GCP eTag which must match the pet SA's current policy. If null, this is ignored.
    * @return The new IAM policy on the user's pet service account, or empty if the eTag value
    *     provided is non-null and does not match current IAM policy on the pet SA.
    */
   public Optional<Policy> enablePetServiceAccountImpersonationWithEtag(
-      UUID workspaceUuid, String userToEnableEmail, String token, @Nullable String eTag) {
-    String petSaEmail =
-        SamRethrow.onInterrupted(
-            () ->
-                samService.getOrCreatePetSaEmail(
-                    gcpCloudContextService.getRequiredGcpProject(workspaceUuid), token),
-            "enablePet");
+      UUID workspaceUuid,
+      String userToEnableEmail,
+      AuthenticatedUserRequest userReq,
+      @Nullable String eTag) {
+    String projectId = gcpCloudContextService.getRequiredGcpProject(workspaceUuid);
+    ServiceAccountName petSaName = getUserPetSa(projectId, userToEnableEmail, userReq).orElse(null);
+    // If the pet SA does not exist, there's no way the eTag can match, so return empty optional.
```

Comment:
Should this comment be on line 108 instead of here?

Reply:
restructured comments to clarify